functions/src/healthSummary/generate.ts

```diff
@@ -0,0 +1,646 @@
+import { jsPDF } from 'jspdf'
+import { type CellDef, type RowInput, type UserOptions } from 'jspdf-autotable'
+import svg2img from 'svg2img'
+import { generateChartSvg } from './generateChart.js'
+import { generateSpeedometerSvg } from './generateSpeedometer.js'
+
+export async function generateHealthSummary(
+  data: HealthSummaryData,
+): Promise<Uint8Array> {
+  const generator = new HealthSummaryPDFGenerator(data)
+  await generator.addFirstPage()
+  await generator.addSecondPage()
+  return generator.finish()
+}
+
+interface TextStyle {
+  fontName: string
+  fontStyle: string
+  fontWeight?: string
+  fontSize: number
+  color?: [number, number, number]
+}
+
+class HealthSummaryPDFGenerator {
```

Comment:
Would also be great if we can use this opportunity to add some very first documentation (might just be additions in the README for now) to start documenting the content of this repo.

Reply:
I will create an issue for that.